MentorMatchmaker/EmailGeneration.fs

```diff
@@ -0,0 +1,149 @@
+﻿
+[<RequireQualifiedAccess>]
+module MentorMatchmaker.EmailGeneration
+
+open MentorMatchmaker
+open System.Net.Mail
+
+[<AutoOpen>]
+module private Implementation =
+    let LengthOfMentorshipInWeeks = 8 // TODO: I need a a way to retrieve it and to make it not static....
+
+    let rec topicName topic =
+        match topic with
+        | IntroductionToFSharp -> "Introduction to F#"
+        | DeepDiveInFSharp -> "Deep dive in F#"
+        | ContributeToOpenSource -> "Contribute to open-source"
+        | ContributeToCompiler -> "Contribute to compiler"
+        | WebDevelopment -> "Web development"
+        | DistributedSystems -> "Distributed systems"
+        | DomainModelling -> "Domain modeling"
+        | MobileDevelopment -> "Mobile development"
+        | MachineLearning -> "Machine learning"
+        | UpForAnything -> "I am up for anything"
+        | DesigningWithTypes -> "Designing with types"
+        | MetaProgramming -> "Meta programming"
+
+    let printLocalMeetingTimes offset (meetingTimes: Set<WeekTime>) =
+        meetingTimes
+        |> Set.map (fun time -> time.AddHours(offset))
+        |> toWeekTimeRanges
+        |> Seq.groupBy (fun weekTimeRange -> weekTimeRange.Start.Weekday)
+        |> Seq.map (fun (weekday, weekTimeRanges) ->
+            let aggregatedTimes =
+                weekTimeRanges
+                |> Seq.map (fun range -> range.Start.Time.ToShortTimeString() + "-" + range.End.Time.AddHours(1).ToShortTimeString())
+                |> String.concat ", "
+
+            $"{weekday}: {aggregatedTimes}<br>"
+        )
+        |> String.concat("\t\t\t")
+
+    let formatListOfSharedInterests topics =
+        topics
+        |> Seq.map topicName
+        |> String.concat ", "
+
+    let possessive (name:string) = name + (if name[name.Length - 1] = 's' then "'" else "'s")
+
+    let generateMenteeMentorEmail { Mentor = mentor; Mentee = mentee; MutualTopics = topics; MutualAvailabilities = availabilities } =
+        let mentor = mentor.PersonalInformation
+        let mentee = mentee.PersonalInformation
+        $"
```

Comment:
I think we're missing a template for an unmatched applicant. Do you think that you could add one?

Reply:
I think I'd rather make a new issue apart from this.